File: java/03-bootstrap/src/main/java/com/fResult/bootstrap/common/SpringUtils.java
package com.fResult.bootstrap.common;

import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.util.StringUtils;

public abstract class SpringUtils {
  private SpringUtils() {
    throw new IllegalStateException("This cannot be instantiated");
  }

  public static ConfigurableApplicationContext run(Class<?> componentClass, String profile) {
    final var applicationContext = new AnnotationConfigApplicationContext();

    if (StringUtils.hasText(profile)) {
      applicationContext.getEnvironment().setActiveProfiles(profile);
    }

    applicationContext.register(componentClass);
    applicationContext.refresh();

    applicationContext.start();
    return applicationContext;
  }
}

File: kotlin/03-bootstrap/src/main/kotlin/com/fResult/bootstrap/common/SpringUtils.kt
package com.fResult.bootstrap.common

import org.springframework.context.ConfigurableApplicationContext
import org.springframework.context.annotation.AnnotationConfigApplicationContext
import kotlin.reflect.KClass

class SpringUtils {
  companion object {
    fun run(componentClass: KClass<*>, profile: String): ConfigurableApplicationContext {
      return AnnotationConfigApplicationContext().apply {
        profile.also { environment.setActiveProfiles(profile) }

        register(componentClass.java)
        refresh()
        start()
      }
    }
  }
}
